tests/v20/wfm/frames_analytical_test.py

```diff
@@ -0,0 +1,226 @@
+                                 window_size=window_size,
+                                 start_chopper_padding_t=padding,
+                                 n_choppers=1)
+    source_to_chopper_distance = 5.0 * sc.units.m
+    source_to_pixel_distance = 10.0 * sc.units.m
+    max_gradient = source_to_chopper_distance / (
+        padding)  # 5.0 is source - chopper distance
+    l_edge_predicted = (source_to_pixel_distance /
+                        max_gradient) + (offset + pulse_length)
+    min_gradient = source_to_pixel_distance / (pulse_length + padding +
+                                               window_opening_t)
+    r_edge_predicted = (source_to_pixel_distance / min_gradient) + (
+        offset + pulse_length + padding + window_opening_t)
+    shift_predicted = window_opening_t / 2.0 + offset + pulse_length + padding
+    instrument = _single_chopper_beamline(window_opening_t=window_opening_t,
+                                          pulse_length=pulse_length,
+                                          window_size=window_size,
+                                          phase=phase)
+    frames = frames_analytical(instrument, offset=offset)
+    # Following results best understood sketching simple time-distance
+    # diagram of above.
+    assert allclose(
+        frames['left_edges'].data,
+        sc.broadcast(dims=['frame'], shape=[1], x=l_edge_predicted))
+    assert allclose(
+        frames['right_edges'].data,
+        sc.broadcast(dims=['frame'], shape=[1], x=r_edge_predicted))
+    assert allclose(frames['shifts'].data,
+                    sc.broadcast(dims=['frame'], shape=[1], x=shift_predicted))
+    return frames
+
+
+def test_frames_analytical_one_chopper_one_cutout():
+    offset = 0.0 * sc.units.us  # start of pulse t offset from 0
+    window_opening_t = 0.5 * sc.units.us
+    pulse_length = 10.0 * sc.units.us
+    window_size = (np.pi / 4.0) * sc.units.rad  # 45 degree opening
+
+    padding = 1.0 * sc.units.us
+    _test_single_chopper_single_cutout(offset, padding, pulse_length,
+                                       window_opening_t, window_size)
+
+
+def test_frames_analytical_one_chopper_one_cutout_different_pulse_offset():
+    # Offsetting the pulse start time should impact directly on the
+    # frame edges and shifts
+    offset = 5.0 * sc.units.us  # start of pulse t offset from 0
+    window_opening_t = 0.5 * sc.units.us
+    pulse_length = 10.0 * sc.units.us
+    window_size = (np.pi / 4.0) * sc.units.rad  # 45 degree opening
+
+    padding = 1.0 * sc.units.us
+    a = _test_single_chopper_single_cutout(offset, padding, pulse_length,
+                                           window_opening_t, window_size)
+    # Now with new larger offset
+    offset += 1.0 * sc.units.us
+    b = _test_single_chopper_single_cutout(offset, padding, pulse_length,
+                                           window_opening_t, window_size)
+    assert sc.all(b['left_edges'].data > a['left_edges'].data).value
```

Comment:
```suggestion
    assert allclose(b['left_edges'].data, a['left_edges'].data + 1.0 * sc.units.us).value
```
and the same for the lines below?